tests/shell/tests.py

```diff
@@ -29,3 +39,17 @@ def test_stdin_read(self, select):
             stdin.seek(0)
             call_command('shell')
         self.assertEqual(stdout.getvalue().strip(), '100')
+
+    @unittest.skipIf(IPython is not None, 'IPython is installed')
+    @mock.patch('django.core.management.commands.shell.select.select')
```

Comment:
Leave a comment about your finding with parallel tests so future test authors for `shell` don't have to re-learn it all. Perhaps even extract the patching into `setUp()` / `tearDown()` of the whole test class so it's automatic